hack/local-up-cluster.sh

```diff
@@ -44,6 +32,57 @@ RoleARN="$NODE_ROLE_ARN"
 EOF
 }
 
-write_cloudconfig
+CLOUD_PROVIDER_ROOT=${CLOUD_PROVIDER_ROOT:-"$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )"}
+KUBE_ROOT=${KUBE_ROOT:-"$GOPATH/src/k8s.io/kubernetes"}
+KUBE_VERSION=${KUBE_VERSION:-v1.22.1}
+
+# Configuration for k8s.io/kubernetes/hack/local-up-cluster.sh
+NODE_ROLE_ARN=${NODE_ROLE_ARN:-""}
+CLOUD_PROVIDER=aws
+EXTERNAL_CLOUD_PROVIDER=true
+CLOUD_CONFIG=${CLOUD_CONFIG:-${CLOUD_PROVIDER_ROOT}/cloudconfig}
+EXTERNAL_CLOUD_PROVIDER_BINARY=${EXTERNAL_CLOUD_PROVIDER_BINARY:-"$GOPATH/src/k8s.io/cloud-provider-aws/aws-cloud-controller-manager"}
+NODE_ZONE=${AWS_NODE_ZONE:-"us-west-2a"}
+NET_PLUGIN="${NET_PLUGIN:-kubenet}"
+CLOUD_CTLRMGR_FLAGS=${CLOUD_CTLRMGR_FLAGS:-"--controllers=*"}
+
+info "\n + + Testing on a local cluster + +\n"
+info "Printing configuration:"
+info "CLOUD_PROVIDER_ROOT=$CLOUD_PROVIDER_ROOT"
+info "KUBE_ROOT=$KUBE_ROOT"
+info "KUBE_VERSION=$KUBE_VERSION"
+info "NODE_ROLE_ARN=$NODE_ROLE_ARN"
+info "CLOUD_PROVIDER=$CLOUD_PROVIDER"
+info "EXTERNAL_CLOUD_PROVIDER=$EXTERNAL_CLOUD_PROVIDER"
+info "CLOUD_CONFIG=$CLOUD_CONFIG"
+info "EXTERNAL_CLOUD_PROVIDER_BINARY=$EXTERNAL_CLOUD_PROVIDER_BINARY"
+info "NODE_ZONE=$NODE_ZONE"
+info "NET_PLUGIN=$NET_PLUGIN"
+info "CLOUD_CTLRMGR_FLAGS=$CLOUD_CTLRMGR_FLAGS\n"
 
+read -p "Do you wish to continue? (y/N)?" answer
+case ${answer:0:1} in
+    y|Y )
+    ;;
+    * )
+        echo "Quitting.."
+        exit 0
+    ;;
+esac
+
+export NODE_ROLE_ARN
+export CLOUD_PROVIDER
+export EXTERNAL_CLOUD_PROVIDER
+export CLOUD_CONFIG
+export EXTERNAL_CLOUD_PROVIDER_BINARY
+export NODE_ZONE
+export NET_PLUGIN
+export CLOUD_CTLRMGR_FLAGS
+(
+    cd ${KUBE_ROOT}
+    git checkout ${KUBE_VERSION}
+    make clean
```

Comment:
`make clean` is required in some cases with generated files but I'm not sure exactly when, perhaps this could be improved in the future and not clean every time as it adds significant build time.